src/backend/distributed/shared_library_init.c

```diff
@@ -194,6 +195,16 @@ CreateRequiredDirectories(void)
 static void
 RegisterCitusConfigVariables(void)
 {
+	DefineCustomIntVariable(
+		"citus.node_connection_timeout",
+		gettext_noop("Sets the maximum duration to connect to worker nodes."),
+		NULL,
+		&NodeConnectionTimeout,
+		5000, 1000, 60 * 60 * 1000,
```

Comment:
Is 5 seconds a good default? The only time this would normally happen in a data centre is if the server is overloaded and the listen queue is full, but in that case you'd much prefer clients to give up quickly.